java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -152,32 +152,69 @@ private void registerMinion(String minionId, boolean isSaltSSH,
                 return false;
             },
             machineId -> {
+                boolean isReactivation = false;
+                ActivationKey ak = null;
+                ValueMap grains = new ValueMap(SALT_SERVICE.getGrains(minionId).orElseGet(HashMap::new));
+                Optional<String> managmentKeyLabel = getManagementKeyLabelFromGrains(grains);
+                if (managmentKeyLabel.isPresent()) {
```

Comment:
can use `ifPresent`

Reply:
No, does not work as I change local variables. This does not compile.